modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java

```diff
@@ -217,10 +217,16 @@ public void testDeleteByQueryOnReadOnlyIndex() throws Exception {
         String block = randomFrom(SETTING_READ_ONLY, SETTING_READ_ONLY_ALLOW_DELETE);
         try {
             enableIndexBlock("test", block);
+            if (block.equals(SETTING_READ_ONLY_ALLOW_DELETE)) {
+                setDiskAllocationDeciderEnabled(false);
+            }
             assertThat(deleteByQuery().source("test").filter(QueryBuilders.matchAllQuery()).refresh(true).get(),
                     matcher().deleted(0).failures(docs));
```

Comment:
We should in the disk decider disabled case change the retry policy for the delete by query request to ensure the test runs faster. I suggest to do: maxRetries=2, retryBackoffInitialTime=50ms in that case. The other cases should just run with defaults.

Reply:
I have added the two parameters yet.